sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -416,10 +416,11 @@ public String getEventHubName() {
         }
 
         final Context finalSharedContext = sharedContext != null ? sharedContext : Context.NONE;
-
         return withRetry(
             getSendLink(batch.getPartitionId()).flatMap(link -> {
-                if (isTracingEnabled) {
+                // if parent context already has send span context data (in case of retries),
+                // don't start a new send span
+                if (isTracingEnabled && !parentContext.get().getData(HOST_NAME_KEY).isPresent()) {
```

Comment:
does it mean we'll send one span for the first try only? 
It should be one span per Send call. If there is more than one try - it should wrap all of them:
- duration should be duration of send() call 
- result is the result of all tries (i.e. eventually sent or failed)

Reply:
> it should wrap all of them

does this mean we want to add links between all the send spans but start only a single span?

Reply:
We still create span per message and link all of these spans to the single 'send' span